plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OraclePoolConnectionFactory.java

```diff
@@ -80,6 +80,10 @@ public OraclePoolConnectionFactory(
     public Connection openConnection(ConnectorSession session)
             throws SQLException
     {
-        return dataSource.getConnection();
+        Connection connection = dataSource.getConnection();
+        // Oracle's pool doesn't reset autocommit of connections when reusing them so we explicitly enable autocommit
+        // by default to match the JDBC spec.
+        connection.setAutoCommit(true);
```

Comment:
Can the oracle pool be configured to do that?

Reply:
Yes, by passing the connection property `OracleConnection.CONNECTION_PROPERTY_AUTOCOMMIT` to `true` to the PoolDataSource.
I'll do that.

Reply:
Not quite - turns out that it only affects the initial connection. UCP isn't touching the connections once created.